### PR TITLE
add Timestamping= to socket units and some other .socket unit tweaks

### PR DESCRIPTION
**Pull request by poettering**

### CODE REVIEW

File: man/systemd.socket.xml [hunk 1]
```diff
@@ -285,12 +284,11 @@
...
         <term><varname>ListenMessageQueue=</varname></term>
-        <listitem><para>Specifies a POSIX message queue name to listen
-        on. This expects a valid message queue name (i.e. beginning
-        with /). Behavior otherwise is very similar to the
-        <varname>ListenFIFO=</varname> directive above. On Linux
-        message queue descriptors are actually file descriptors and
-        can be inherited between processes.</para></listitem>
+        <listitem><para>Specifies a POSIX message queue name to listen on (see <citerefentry
+        project='man-pages'><refentrytitle>mq_overview</refentrytitle><manvolnum>7</manvolnum></citerefentry>
+        for details). This expects a valid message queue name (i.e. beginning with /). Behavior otherwise is
```
Comment (keszybz) [inline man/systemd.socket.xml]: `<literal>/</>`

File: src/core/socket.c [hunk 2]
```diff
@@ -3409,6 +3422,39 @@ static const char* const socket_result_table[_SOCKET_RESULT_MAX] = {
 
 DEFINE_STRING_TABLE_LOOKUP(socket_result, SocketResult);
 
+static const char* const socket_timestamping_table[_SOCKET_TIMESTAMPING_MAX] = {
+        [SOCKET_TIMESTAMPING_OFF] = "off",
+        [SOCKET_TIMESTAMPING_USEC] = "usec",
+        [SOCKET_TIMESTAMPING_NSEC] = "nsec",
```
Comment (keszybz) [inline src/core/socket.c]: `nsec` is a pretty bad name. The actual unit name is 'nanosecond' and the only official abbreviation is `ns`. "sec" and "nsec" and such are neither here nor there.

**Review by keszybz — COMMENTED** [review]
Looks good. Maybe the abbreviation could be changed though.

**Comment by poettering** [comment]
Force pushed a new version, only two changes are the suggested ones. ptal

**Comment by keszybz** [comment]
LGTM.

**Comment by keszybz** [comment]
bionic-amd64 failed in TEST-50-DISSECT.
semaphoreci timed out while downloading packages.

**Pull request merged**